fs/fs_zenfs.cc

```diff
@@ -1096,6 +1485,98 @@ Status ZenFS::MkFS(std::string aux_fs_path, uint32_t finish_threshold,
   return Status::OK();
 }
 
+
+Status ZenFS::MkFSV2(std::string aux_fs_path, uint32_t finish_threshold,
+                   uint32_t max_open_limit, uint32_t max_active_limit) {
+  std::vector<Zone*> metazones = zbd_->GetMetaZones();
+  std::vector<Zone*> snapshot_zones = zbd_->GetSnapshotZones();
+  std::unique_ptr<ZenMetaLog> metadata_log;
+  std::unique_ptr<ZenMetaLog> snapshot_log;
+  Zone* meta_zone = nullptr;
+  Zone* snapshot_zone = nullptr;
+  IOStatus s;
+
+  /* TODO: check practical limits */
+
+  if (max_open_limit > zbd_->GetMaxOpenZones()) {
+    return Status::InvalidArgument(
+        "Max open zone limit exceeds the device limit\n");
+  }
+
+  if (max_active_limit > zbd_->GetMaxActiveZones()) {
+    return Status::InvalidArgument(
+        "Max active zone limit exceeds the device limit\n");
+  }
+
+  if (max_active_limit < max_open_limit) {
+    return Status::InvalidArgument(
+        "Max open limit must be smaller than max active limit\n");
+  }
+
+  if (aux_fs_path.length() > 255) {
+    return Status::InvalidArgument(
+        "Aux filesystem path must be less than 256 bytes\n");
+  }
+
+  ClearFiles();
+  zbd_->ResetUnusedIOZones();
+
+  for (const auto mz : metazones) {
+    if (mz->Reset().ok()) {
+      if (!meta_zone) meta_zone = mz;
+    } else {
+      Warn(logger_, "Failed to reset meta zone\n");
+    }
+  }
+
+  if (!meta_zone) {
+    return Status::IOError("No available meta zones\n");
+  }
+  metadata_log.reset(new ZenMetaLog(zbd_, meta_zone));
+
+  Superblock* metadata_super = new Superblock(zbd_, aux_fs_path, finish_threshold,
+                                     max_open_limit, max_active_limit);
+  std::string metadata_super_string;
+  metadata_super->EncodeTo(&metadata_super_string);
+
+  // write an empty superblock to meta zone
+  s = metadata_log->AddRecord(metadata_super_string);
+  if (!s.ok()) return std::move(s);
+
+  for (const auto sz: snapshot_zones) {
+    if (sz->Reset().ok()) {
+      if (!snapshot_zone) snapshot_zone = sz;
+    } else {
+      Warn(logger_, "Failed to reset snapshot zone\n");
+    }
+  }
+
+  if (!snapshot_zone) {
+    return Status::IOError("No available snapshot zones\n");
+  }
+
+  snapshot_log.reset(new ZenMetaLog(zbd_, snapshot_zone));
+
+  Superblock* snapshot_super = new Superblock(zbd_, aux_fs_path, finish_threshold,
+                                     max_open_limit, max_active_limit);
+  std::string snapshot_super_string;
+  snapshot_super->EncodeTo(&snapshot_super_string);
+
+  // write an empty superblock to snapshot zone
+  s = snapshot_log->AddRecord(snapshot_super_string);
```

Comment:
The logic here could be merged with the previous for loop, we can directly change the original MkFS and extract the `metadata` zone for loop to a new helper function, this new help function could take parameter to persistent both op log zone and snapshot zone